elastic/integrations
# [rubrik] Add managed volumes datastream

## Proposed commit message
The PR adds a new data stream to collect the metrics for Managed Volumes. Managed Volumes provide hosts with a backup target location on a Rubrik cluster. Following are the metrics collected:
| Metric                    | Description                                                                                           |
|---------------------------|-------------------------------------------------------------------------------------------------------|
| Free Size                 | Free capacity for the volume across all the channels, in bytes.                                      |
| Num Channels              | Number of channels to divide the volume into. Each channel provides a unique share to write to.      |
| Pending Snapshot Count    | Pending snapshots.                                                                                   |
| Snapshot Count            | Number of snapshots.                                                                                 |
| State                     | State of managed volume.                                                                             |
| Used Size                 | Used capacity for the volume across all the channels, in bytes.                                      |
| Volume Size               | Maximum capacity for the volume across all the channels, in bytes.                                   |



### Note for reviewers:
Please ignore the fact the fact the changes don't contain the docs yet. That will be added for the entire package later once we have a few data streams merged. 

## Checklist


## Author's Checklist



## How to test this PR locally
- Spin up the Elastic stack and point your `elastic-package` to the running Kibana instance
- Build the package locally and install the package using `elastic-package install --zip rubrik-0.1.0.zip`
- Observe the metrics in Kibana > Discovers section under the `rubrik_managed_volumes` dataset

## Related issues
- Closes https://github.com/elastic/obs-integration-team/issues/123


## Screenshots
![image](https://github.com/user-attachments/assets/4496854e-a3ee-44c0-90be-999b86a3a3e4)
